src/session.cc

```diff
@@ -181,6 +193,150 @@ Config::debug_log(const char* fmt, ...) {
   }
 }
 
+bool
+getPublicKeyAttribute(Config& config_, CK_ATTRIBUTE_PTR pAttribute)
```

Comment:
Trailing underscore is used here for member variables, not function parameters.

(also below)